Lib/argparse.py

```
@@ -2572,24 +2576,38 @@ def _get_value(self, action, arg_string):
         # return the converted value
         return result
 
-    def _check_value(self, action, value):
+    def _check_value(self, action, value, arg_string=None):
         # converted value must be one of the choices (if specified)
         choices = action.choices
         if choices is None:
             return
 
+        if arg_string is None:
+            arg_string = value
+
         if isinstance(choices, str):
             choices = iter(choices)
 
-        if value not in choices:
-            args = {'value': str(value),
+        typed_choices = []
+        if (self.convert_choices and
+            action.type and
+            all(isinstance(choice, str) for choice in choices)
+        ):
+            try:
+                typed_choices = [action.type(v) for v in choices]
+            except Exception:
+                # We use a blanket catch here, because type is user provided.
+                pass
```

Silently ignoring is not the right way to go about this.